src/DeloitteDigital.Atlas/Mvc/AdvancedModelLocator.cs

```diff
@@ -45,7 +44,7 @@ protected override object GetModelFromTypeName(string typeName, string model, bo
             }
 
             // Get model instance from service location and resolve the instance's dependencies
-            var modelObject = ServiceLocator.Current.GetInstance(modelType);
+            var modelObject = Sitecore.DependencyInjection.ServiceLocator.ServiceProvider.GetService(modelType);
```

Comment:
This should use `TypeHelper.CreateObject(modelType);` since we do not want to have to register all view model types in advance.

Reply:
It should work according to this: http://jockstothecore.com/customizing-the-model-within-sitecore-mvc/ 
Note - the original ModelLocator might actually work with generics too?! So the AdvancedModelLocator might no longer be required.

Reply:
@heikof @geekrocket what works for me is as follow:
```
var modelObject = Sitecore.DependencyInjection.ServiceLocator.ServiceProvider.GetService(modelType);
if (modelObject == null)
{
    modelObject = Activator.CreateInstance(modelType);
}
```
The null check and instance creation is for not having to register static types used by Sitecore in SPEAK views.

Reply:
@heikof I tested the original ModelLocator in Sitecore9, still does not work with generics. The AdvancedModelLocator serves three purposes (1) to resolve rendering model types without having to register them in Sitecore as part of the view rendering item, (2) to be able to resolve generic rendering model types, and (3) to resolve dependencies of the rendering models.

Reply:
Thanks @davidnguyen 
Correct me if I am wrong, but the code you have posted above (or my suggestion for that matter) does not solve use case (3). Calling `Activator.CreateInstance(modelType)` won't resolve any dependencies that the view model might have. 
I think the only option we would have in that case is to use the Sitecore DI and register all view models :( other ideas?

Reply:
@heikof true, that one is just the fall back for resolving types that are not registered with the container, and that the container library does not provide any clear way to resolve unregistered types. That definitely won't resolve any dependencies. Integrating with Sitecore DI is a much better option.